### ISSUE TITLE
Waveform not in sync with MediaElement backend audio when zooming

### ISSUE THREAD

**Issue by seklum** [wavesurfer-js/wavesurfer.js#1714]
When zooming in enough it's clear that the majority of beats are heard before the actual section of the waveform is hit. It's a small amount but noticeable for the the project I am using it for. It appears that the audio is 0.04 seconds ahead of the progress bar. The song I am using is about 3 minutes long if that makes any difference.

Wavesurfer.js Version 3.0.0

Firefox Developer Version 69.0b9 (64-bit)
Windows 10

```html
<script>var wavesurfer = WaveSurfer.create({
	container: '#waveform',
	waveColor: '#222222',
	progressColor: 'orange',
	backend: 'MediaElement',
	plugins: [
		WaveSurfer.cursor.create({
			showTime: true,
			opacity: 1,
			color: 'blue',
			customShowTimeStyle: {
				'background-color': '#000',
				color: '#fff',
				padding: '2px',
				'font-size': '10px'
			}
		})
	]
});
wavesurfer.load(song);


//Zoom slider
var slider = document.querySelector('[data-action="zoom"]');

slider.value = wavesurfer.params.minPxPerSec;
slider.min = wavesurfer.params.minPxPerSec;

slider.addEventListener('input', function() {
	wavesurfer.zoom(Number(this.value));
})

//Set initial zoom to match slider value
wavesurfer.zoom(slider.value);

function playPause() {
	wavesurfer.playPause();
}</script>

<div id="waveform"></div>
<input data-action="zoom" type="range" min="20" max="200" value="0" style="width: 100%">
<button onClick="playPause()">Play/Pause</button>
```

I can reproduce it with any of the songs that I currently have set up with my project.

**Comment by thijstriemstra** [comment]
can you also reproduce it with older wavesurfer.js versions, e.g. 2.2.1?

**Comment by seklum** [comment]
Yes I'm getting the same thing with version 2.2.1. The only way to really see it is to zoom in as much as possible and then place the progress indicator at the exact start of a beat and then hit play. Once you move it back a fraction of a second you hear the actual spike that your trying to view on the waveform.

**Comment by thijstriemstra** [comment]
Does it happen with the default webaudio backend?

**Comment by seklum** [comment]
No this error does not occur when I comment out the backend parameter.

**Comment by thijstriemstra** [comment]
can you narrow it down to specific code in `MediaElement` backend?

**Comment by seklum** [comment]
I'll give it a try. I've only been working with js for a week or two so I may not be able to solve it just yet.

**Comment by thijstriemstra** [comment]
Start with https://github.com/katspaugh/wavesurfer.js#development

**Comment by seklum** [comment]
Ok so far I've at least looked over all the code in the mediaelement file. I can't see anything that looks like it would cause this. I've verified that the waveforms are not different they still display the same so it's something with the audio only.

Here is one interesting thing I was able to find. At the end of the waveform when using the mediaelement backend the cursor extends a .02 seconds past the edge of the window extending it. This does not occur with the default backend. I don't know how to check the beginning but it appears that mediaelement has some sort of different duration value. It does appear to be approximately .04 which could be related to the .02 second extended duration at the end.

The audio is being clipped out for the first .04 seconds of the file and and extra .02s added to the end.

**Comment by thijstriemstra** [comment]
does it happen with just that audio file or all?

**Comment by NegiBaba** [comment]
Hey guys, I seem to be a bit late on using this library but I came across the same issue. 
I am recording an audio using react-mic library and passing the audio blob to the wavesurfer to generate waveform and play the audio. **But!** when I seeking to more than 3/4th of the audio in waveform the seeker overflows and scrollbar gets rendered below the waveform, even though I have set scrollParent false and responsive: true.
